Pod/Classes/SEGAmplitudeIntegration.m

```diff
@@ -242,7 +242,7 @@ - (void)incrementOrSetTraits:(NSDictionary *)traits
         } else if ([self.traitsToSetOnce member:trait]) {
             [self.amplitude identify:[self.identify setOnce:trait value:value]];
         } else {
-            [self.amplitude identify:[self.identify set:trait value:value]];
+            [self.amplitude setUserProperties:@{trait: value}];
```

Comment:
The `setUserProperties` will create a new `AMPIdentify` then identify with it, while `identify:[self.identify set:trait value:value]` will first call set to the current identify, that's when the check and log `AMPLITUDE_LOG(@"Already used property '%@' in previous operation, ignoring for operation '%@'", property, operation);` kick in, so the user properties not updated. 

@cerupcat I see the description in the issue, but the change here is not for either `traitsToIncrement` or `traitsToSetOnce`, would you mind share a bit more about the user case to overwrite the user property? One thing for this change is `self.identify` would not be in sync with the newest value though.

Reply:
Thanks for the response.

Here's a bit more context. I'm trying to set or update (if already set) user properties by calling `analytics.identify(nil, traits: [name: value])`.  Some of the traits that are set are setOnce, but most are not. The setOnce traits appear to work correctly. However, I notice when setting a trait that is not setOnce, the new value is not updated in Amplitude; only the first value is set (or updated) when the app starts - subsequent values that are set during runtime are thrown out as "already used property."

What I expect to happen would be every time a value is set for a trait, that value should be set/updated on the user, but that's not what happens.

When not using any setOnce or traitsToIncrement, this integration uses `[self.amplitude setUserProperties:@{trait: value}]` instead of `[self.amplitude identify:[self.identify set:trait value:value]].` Using `[self.amplitude setUserProperties:@{trait: value}]` does do as expected in terms of updating the value for a given property in amplitude.

Is there an alternative way of updating user property values?

Reply:
@cerupcat I just take a look at `AMPIdentify`, basically for the `AMPIdentify` object, inside one object it seems the same property will be set only once, and the payload will be sent to server, and server will handle all the `set` and `setOnce` logics, and inside one identify object it just support one property once.

So the case here is there is some properties is in traitsToSetOnce, and some others are not, and other properties have updated values during running. Sounds like a valid case then. Originally I was considering if the idenify object inside class already have the property to set, then setUserProperies may be overwrittem later by the previous value. But if we just change this line here, I guess there is no more set operation in this identify object. And seems anyway there is an identify event now so I guess that looks fine for me. The alternatives would be empty traitsToSetOnce or set a new AMPIdentify object to `self.identify`, that may save some idenitfy events but would need a bit extra work out of library.